### PR TITLE
[Editorial] Xor Pyramid

### PR DESCRIPTION
**Pull request by brebenelmihnea**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: content/3_Silver/Intro_Bitwise.mdx [hunk 1]
```diff
@@ -341,6 +341,75 @@ def prod(a: int, b: int) -> int:
...
+Pascal's triangle of height five are: $1,4,6,4,1$. We can think of these values as the number of occurrences of each bottom row value.
+In the above image, value $A$ will be xored one time in the result, while value $B$ will be xored four times in the result. Therefore,
+the parity of the frequency tells us that value $A$ contributes to the result, whil value $B$ is cancels out itself.
+
+
+### Implementation
+
+Because we are only interested in the parity of the binomial coefficient, there is no need of computing its actual value. We can simply check the parity by
+counting the power of two in $\binom{n}{k}=\frac{n!}{k! \cdot (n-k)!}$. With this in mind, precompute $pref_i=p$ where $p$ is the power of two in $i!$ factorization.
+Therefore, the binomial coefficient is odd if: $pref_n-pref_k-pref_{n-k}=0$.
```
Comment (SansPapyrus683) [inline content/3_Silver/Intro_Bitwise.mdx:373]: should be in explanation ngl

File: content/3_Silver/Intro_Bitwise.mdx [hunk 2]
```diff
@@ -341,6 +341,75 @@ def prod(a: int, b: int) -> int:
...
+```cpp
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+int main() {
+	int n;
+	cin >> n;
+	vector<int> pref(n + 1, 0);
```
Comment (SansPapyrus683) [inline content/3_Silver/Intro_Bitwise.mdx:389]: , 0 not needed

File: content/3_Silver/Intro_Bitwise.mdx [hunk 3]
```diff
@@ -341,6 +341,75 @@ def prod(a: int, b: int) -> int:
...
+	for (int i = 2; i <= n; i++) {
+		int cnt = i;
```
Comment (SansPapyrus683) [inline content/3_Silver/Intro_Bitwise.mdx:391]: better name

File: content/3_Silver/Intro_Bitwise.mdx [hunk 4]
```diff
@@ -341,6 +341,75 @@ def prod(a: int, b: int) -> int:
...
+	}
+	int ans = 0;
+	for (int i = 0; i < n; i++) {
+		int num;
+		cin >> num;
+		if (pref[n - 1] - pref[i] - pref[n - i - 1] == 0) { ans ^= num; }
+	}
+	cout << ans << '\n';
+
+	return 0;
```
Comment (SansPapyrus683) [inline content/3_Silver/Intro_Bitwise.mdx:406]:
```suggestion

	cout << ans << endl;
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by envyaims — APPROVED** [review]
ok

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**